specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,1430 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Cognitive Language Service - Analyze Conversations",
+    "description": "This API accepts a request and mediates among multiple language projects, such as LUIS Generally Available, Question Answering, Conversation, and then calls the best candidate service to handle the request. At last, it returns a response with the candidate service's response as a payload.\n\n In some cases, this API needs to forward requests and responses between the caller and an upstream service.",
+    "version": "2022-03-01-preview"
+  },
+  "securityDefinitions": {
+    "apim_key": {
+      "type": "apiKey",
+      "description": "An APIM subscription key.",
+      "name": "Ocp-Apim-Subscription-Key",
+      "in": "header"
+    }
+  },
+  "security": [
+    {
+      "apim_key": []
+    }
+  ],
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{Endpoint}/language",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "$ref": "common.json#/parameters/Endpoint"
+      }
+    ]
+  },
+  "paths": {
+    "/:analyze-conversations": {
```

Comment:
Is this API that important to be in root path? Would the service grow and add more APIs when you would like it to be e.g. `/conversations/:analyze`

Reply:
We have a converged plan with other Language teams so that each service only offers a single API - in this case /:analyze-conversations - and the user specifies the type of service or capability they need in the request body. For example, we offer the following conversational tasks: "CustomConversation" and "entityRecognition". The user calls /:analyze-conversations for both tasks and they specify their choice in the body (polymorphically) using the discriminator pattern.